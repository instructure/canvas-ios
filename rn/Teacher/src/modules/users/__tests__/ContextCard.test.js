//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* eslint-disable flowtype/require-valid-file-annotation */

import { shallow } from 'enzyme'
import React from 'react'
import {
  ContextCard,
  props,
} from '../ContextCard'
import app from '../../app'

import renderer from 'react-test-renderer'
import explore from '../../../../test/helpers/explore'

jest.mock('../../../routing/Screen')
  .mock('react-native/Libraries/Components/Touchable/TouchableHighlight', () => 'TouchableHighlight')
  .mock('react-native/Libraries/Components/Touchable/TouchableOpacity', () => 'TouchableOpacity')
  .mock('../../../common/components/ErrorView.js', () => 'ErrorView')

const templates = {
  ...require('../../../__templates__/course'),
  ...require('../../../__templates__/enrollments'),
  ...require('../../../__templates__/users'),
  ...require('../../../__templates__/section'),
  ...require('../../../__templates__/assignments'),
  ...require('../../../__templates__/submissions'),
  ...require('../../../redux/__templates__/app-state'),
  ...require('../../../canvas-api-v2/queries/__templates__/ContextCard'),
}

const user = templates.user({ id: '1', analytics: { tardinessBreakdown: { late: 10, missing: 20 } } })
const defaultProps = {
  courseID: '1',
  userID: '1',
  user,
  course: templates.course({ id: '1' }),
  enrollment: templates.enrollment({ id: '1', course_id: '1', user_id: '1', course_section_id: '32', grades: { current_grade: 'A', current_score: 50 }, section: templates.section() }),
  submissions: [templates.submission({ id: '1', assignment_id: '1', grade: 50, assignment: templates.assignment({ id: '1', points_possible: 100 }) })],
  courseColor: '#fff',
  navigator: { dismiss: jest.fn(), show: jest.fn() },
  refresh: jest.fn(),
  loading: false,
  modal: false,
  permissions: {
    viewAnalytics: true,
    viewAllGrades: true,
    sendMessages: true,
  },
  isStudent: true,
}

beforeEach(() => {
  jest.clearAllMocks()
  app.setCurrentApp('teacher')
})

describe('ContextCard', () => {
  it('renders', () => {
    let view = renderer.create(
      <ContextCard {...defaultProps} />
    )
    expect(view.toJSON()).toMatchSnapshot()
  })

  it('renders last activity in teacher', () => {
    app.setCurrentApp('teacher')
    const view = shallow(<ContextCard {...defaultProps} />)
    const label = view.find('[testID="context-card.last-activity"]')
    expect(label).not.toBeNull()
  })

  it('does not render last activity in student', () => {
    app.setCurrentApp('student')
    const view = shallow(<ContextCard {...defaultProps} />)
    const label = view.find('[testID="context-card.last-activity"]')
    expect(label).not.toBeNull()
  })

  it('renders user name', () => {
    let props = {
      ...defaultProps,
      user: {
        ...defaultProps.user,
        short_name: 'Alfredo',
      },
    }
    let view = shallow(<ContextCard {...props} />)
    let header = shallow(view.find('FlatList').prop('ListHeaderComponent'))
    let label = header.find('[testID="ContextCard.userNameLabel"]')
    expect(label.prop('children')).toEqual('Alfredo')
  })

  it('renders user pronouns', () => {
    let props = {
      ...defaultProps,
      user: {
        ...defaultProps.user,
        short_name: 'Alfredo',
        pronouns: 'He/Him',
      },
    }
    let view = shallow(<ContextCard {...props} />)
    let header = shallow(view.find('FlatList').prop('ListHeaderComponent'))
    let label = header.find('[testID="ContextCard.userNameLabel"]')
    expect(label.prop('children')).toEqual('Alfredo (He/Him)')
  })

  it('renders for a user that cannot view analytics', () => {
    let view = renderer.create(
      <ContextCard {...defaultProps} permissions={{ ...defaultProps.permissions, viewAnalytics: false }} isStudent={false} />
    )
    expect(view.toJSON()).toMatchSnapshot()
  })

  it('renders for a user that cannot send messages', () => {
    var view = renderer.create(
      <ContextCard {...defaultProps} permissions={{ ...defaultProps.permissions, sendMessages: false }} isStudent={false} />
    )
    expect(view.toJSON()).toMatchSnapshot()
    view = renderer.create(
      <ContextCard {...defaultProps} permissions={{ ...defaultProps.permissions, sendMessages: false }} />
    )
    expect(view.toJSON()).toMatchSnapshot()
  })

  it('shows the activity indicator when pending', () => {
    let view = renderer.create(
      <ContextCard {...defaultProps} loading={true} course={null} />
    )

    expect(view.toJSON()).toMatchSnapshot()
  })

  it('formats the last_activity_at properly', () => {
    let enrollment = templates.enrollment({
      id: '1',
      course_id: '1',
      user_id: '1',
      course_section_id: '32',
      last_activity_at: '2017-04-05T15:12:45Z',
      grades: {
        current_grade: '100',
      },
      section: templates.section(),
    })
    let view = renderer.create(
      <ContextCard {...defaultProps} enrollment={enrollment} />
    )

    expect(view.toJSON()).toMatchSnapshot()
  })

  it('shows points values when there is no grade', () => {
    let enrollment = templates.enrollment({
      id: '1',
      course_id: '1',
      user_id: '1',
      course_section_id: '32',
      last_activity_at: '2017-04-05T15:12:45Z',
      grades: {
        current_score: 100,
      },
      section: templates.section(),
    })

    let view = renderer.create(
      <ContextCard {...defaultProps} enrollment={enrollment} />
    )

    expect(view.toJSON()).toMatchSnapshot()
  })

  it('renders for a non student', () => {
    let enrollment = templates.enrollment({
      id: '1',
      course_id: '1',
      user_id: '1',
      course_section_id: '32',
      type: 'TeacherEnrollment',
      section: templates.section(),
    })

    let view = renderer.create(
      <ContextCard {...defaultProps} enrollment={enrollment} isStudent={false} />
    )

    expect(view.toJSON()).toMatchSnapshot()
  })

  it('renders error if an error occured', () => {
    let view = renderer.create(
      <ContextCard {...defaultProps} error={Error('oh no an error happened')} />
    )

    expect(view.toJSON()).toMatchSnapshot()
  })

  it('renders if there is no enrollment', () => {
    let view = renderer.create(
      <ContextCard {...defaultProps} enrollment={null} />
    )

    expect(view.toJSON()).toMatchSnapshot()
  })

  it('renders if there is no section in the enrollment', () => {
    let enrollment = templates.enrollment({ id: '1', course_id: '1', user_id: '1', course_section_id: '32' })
    let view = renderer.create(
      <ContextCard {...defaultProps} enrollment={enrollment} />
    )

    expect(view.toJSON()).toMatchSnapshot()
  })

  it('navigate to speedgrader', () => {
    let view = renderer.create(
      <ContextCard {...defaultProps} />
    )
    let assignmentID = defaultProps.submissions[0].assignment.id
    let row = explore(view.toJSON()).selectByID(`user-submission-row.cell-${assignmentID}`)
    expect(row).not.toBeNull()
    row && row.props.onPress()

    expect(defaultProps.navigator.show).toHaveBeenCalled()
  })

  it('navigates to composer', () => {
    defaultProps.navigator.show = jest.fn()
    let view = renderer.create(
      <ContextCard {...defaultProps} />
    )
    const tree = view.toJSON()
    const mailButton: any = explore(tree).selectRightBarButton('context-card.email-contact')
    expect(mailButton).not.toBeNull()
    mailButton.action()
    let expectedProps = { 'canSelectCourse': false, 'contextCode': `course_${defaultProps.course.id}`, 'contextName': `${defaultProps.course.name}`, 'recipients': [user] }
    expect(defaultProps.navigator.show).toHaveBeenCalledWith(`/conversations/compose`, { 'modal': true }, expectedProps)
  })
})

describe('props', () => {
  it('should parse the props correctly from graphql data', () => {
    const data = templates.ContextCardResult()
    const result = props({ data })
    expect(result.course).toMatchObject(data.course)
    expect(result.user).toMatchObject(data.course.users.edges[0].user)
  })

  it('should parse the props correctly in an error case', () => {
    const data = { error: Error('there was an error run for the hills') }
    const result = props({ data })
    expect(result.error).toMatchObject(data.error)
  })

  it('should parse the props and get the loading state correctly', () => {
    const data = templates.ContextCardResult()
    data.course = null
    data.loading = true
    const result = props({ data })
    expect(result).toMatchObject({ loading: true })
  })

  it('should parse the props and if user or course is missing it should bail', () => {
    const data = templates.ContextCardResult()
    data.course = null
    const result = props({ data })
    expect(result.error).toBeDefined()
  })
})
