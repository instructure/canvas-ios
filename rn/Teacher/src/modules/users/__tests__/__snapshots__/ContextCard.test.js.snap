// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextCard formats the last_activity_at properly 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list>
    <item>
      <TouchableHighlight
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }
        testID="user-submission-row.cell-1"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 54,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
            testID="user-submission-row-icon-published-1.icon-img"
          >
            <View
              accessibilityLabel="Published"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/images/course/Assignment.png",
                  }
                }
                style={
                  Array [
                    false,
                    Object {
                      "minHeight": 24,
                      "minWidth": 24,
                    },
                  ]
                }
                testID="access-icon-image"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "bottom": -6,
                    "height": 16,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": -5,
                    "width": 16,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/publish-small.png",
                    }
                  }
                  style={
                    Object {
                      "height": 12,
                      "tintColor": "#00AC18",
                      "width": 12,
                    }
                  }
                  testID="access-icon-icon"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontWeight": "600",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              learn to write code
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "paddingTop": 2,
                    },
                    Object {
                      "marginBottom": 8,
                    },
                    Object {
                      "color": "#8B969E",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              Not Submitted
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    </item>
  </list>
</Screen>
`;

exports[`ContextCard renders 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list>
    <item>
      <TouchableHighlight
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }
        testID="user-submission-row.cell-1"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 54,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
            testID="user-submission-row-icon-published-1.icon-img"
          >
            <View
              accessibilityLabel="Published"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/images/course/Assignment.png",
                  }
                }
                style={
                  Array [
                    false,
                    Object {
                      "minHeight": 24,
                      "minWidth": 24,
                    },
                  ]
                }
                testID="access-icon-image"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "bottom": -6,
                    "height": 16,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": -5,
                    "width": 16,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/publish-small.png",
                    }
                  }
                  style={
                    Object {
                      "height": 12,
                      "tintColor": "#00AC18",
                      "width": 12,
                    }
                  }
                  testID="access-icon-icon"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontWeight": "600",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              learn to write code
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "paddingTop": 2,
                    },
                    Object {
                      "marginBottom": 8,
                    },
                    Object {
                      "color": "#8B969E",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              Not Submitted
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    </item>
  </list>
</Screen>
`;

exports[`ContextCard renders error if an error occured 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
>
  <ErrorView
    error={[Error: oh no an error happened]}
  />
</Screen>
`;

exports[`ContextCard renders for a non student 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list />
</Screen>
`;

exports[`ContextCard renders for a user that cannot send messages 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list />
</Screen>
`;

exports[`ContextCard renders for a user that cannot send messages 2`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={Array []}
  subtitle="Learn React Native"
  title="The Donald"
>
  <list>
    <item>
      <TouchableHighlight
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }
        testID="user-submission-row.cell-1"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 54,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
            testID="user-submission-row-icon-published-1.icon-img"
          >
            <View
              accessibilityLabel="Published"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/images/course/Assignment.png",
                  }
                }
                style={
                  Array [
                    false,
                    Object {
                      "minHeight": 24,
                      "minWidth": 24,
                    },
                  ]
                }
                testID="access-icon-image"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "bottom": -6,
                    "height": 16,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": -5,
                    "width": 16,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/publish-small.png",
                    }
                  }
                  style={
                    Object {
                      "height": 12,
                      "tintColor": "#00AC18",
                      "width": 12,
                    }
                  }
                  testID="access-icon-icon"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontWeight": "600",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              learn to write code
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "paddingTop": 2,
                    },
                    Object {
                      "marginBottom": 8,
                    },
                    Object {
                      "color": "#8B969E",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              Not Submitted
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    </item>
  </list>
</Screen>
`;

exports[`ContextCard renders for a user that cannot view analytics 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list />
</Screen>
`;

exports[`ContextCard renders if there is no enrollment 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list>
    <item>
      <TouchableHighlight
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }
        testID="user-submission-row.cell-1"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 54,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
            testID="user-submission-row-icon-published-1.icon-img"
          >
            <View
              accessibilityLabel="Published"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/images/course/Assignment.png",
                  }
                }
                style={
                  Array [
                    false,
                    Object {
                      "minHeight": 24,
                      "minWidth": 24,
                    },
                  ]
                }
                testID="access-icon-image"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "bottom": -6,
                    "height": 16,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": -5,
                    "width": 16,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/publish-small.png",
                    }
                  }
                  style={
                    Object {
                      "height": 12,
                      "tintColor": "#00AC18",
                      "width": 12,
                    }
                  }
                  testID="access-icon-icon"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontWeight": "600",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              learn to write code
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "paddingTop": 2,
                    },
                    Object {
                      "marginBottom": 8,
                    },
                    Object {
                      "color": "#8B969E",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              Not Submitted
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    </item>
  </list>
</Screen>
`;

exports[`ContextCard renders if there is no section in the enrollment 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list>
    <item>
      <TouchableHighlight
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }
        testID="user-submission-row.cell-1"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 54,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
            testID="user-submission-row-icon-published-1.icon-img"
          >
            <View
              accessibilityLabel="Published"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/images/course/Assignment.png",
                  }
                }
                style={
                  Array [
                    false,
                    Object {
                      "minHeight": 24,
                      "minWidth": 24,
                    },
                  ]
                }
                testID="access-icon-image"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "bottom": -6,
                    "height": 16,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": -5,
                    "width": 16,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/publish-small.png",
                    }
                  }
                  style={
                    Object {
                      "height": 12,
                      "tintColor": "#00AC18",
                      "width": 12,
                    }
                  }
                  testID="access-icon-icon"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontWeight": "600",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              learn to write code
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "paddingTop": 2,
                    },
                    Object {
                      "marginBottom": 8,
                    },
                    Object {
                      "color": "#8B969E",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              Not Submitted
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    </item>
  </list>
</Screen>
`;

exports[`ContextCard shows points values when there is no grade 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
  subtitle="Learn React Native"
  title="The Donald"
>
  <list>
    <item>
      <TouchableHighlight
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        onPress={[Function]}
        style={
          Array [
            undefined,
            undefined,
          ]
        }
        testID="user-submission-row.cell-1"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "flex": 1,
                "flexDirection": "row",
                "minHeight": 54,
                "paddingBottom": 12,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 12,
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
              }
            }
            testID="user-submission-row-icon-published-1.icon-img"
          >
            <View
              accessibilityLabel="Published"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 0,
                }
              }
            >
              <Image
                resizeMode="cover"
                source={
                  Object {
                    "testUri": "../../../src/images/course/Assignment.png",
                  }
                }
                style={
                  Array [
                    false,
                    Object {
                      "minHeight": 24,
                      "minWidth": 24,
                    },
                  ]
                }
                testID="access-icon-image"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderRadius": 8,
                    "bottom": -6,
                    "height": 16,
                    "justifyContent": "center",
                    "position": "absolute",
                    "right": -5,
                    "width": 16,
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "testUri": "../../../src/images/publish-small.png",
                    }
                  }
                  style={
                    Object {
                      "height": 12,
                      "tintColor": "#00AC18",
                      "width": 12,
                    }
                  }
                  testID="access-icon-icon"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
                Object {
                  "marginLeft": 12,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={2}
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontWeight": "600",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay-semibold",
                  },
                ]
              }
            >
              learn to write code
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                  Object {
                    "color": "#2D3B45",
                    "fontSize": 16,
                  },
                  Array [
                    Object {
                      "fontSize": 14,
                      "paddingTop": 2,
                    },
                    Object {
                      "marginBottom": 8,
                    },
                    Object {
                      "color": "#8B969E",
                    },
                  ],
                  Object {
                    "fontFamily": ".SFUIDisplay",
                  },
                ]
              }
            >
              Not Submitted
            </Text>
          </View>
        </View>
      </TouchableHighlight>
    </item>
  </list>
</Screen>
`;

exports[`ContextCard shows the activity indicator when pending 1`] = `
<Screen
  navBarColor="#fff"
  navBarStyle="dark"
  rightBarButtons={
    Array [
      Object {
        "accessibilityLabel": "Send message",
        "action": [Function],
        "image": Object {
          "testUri": "../../../src/images/mail-small.png",
        },
        "testID": "context-card.email-contact",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="small"
      style={
        Object {
          "flex": 1,
        }
      }
    />
  </View>
</Screen>
`;
